src/VisualStudio/Core/Def/Implementation/Watson/WatsonReporter.cs

```diff
@@ -114,6 +114,39 @@ public static void ReportNonFatal(Exception exception)
             session.PostEvent(faultEvent);
         }
 
+        private static string GetDescription()
+        {
+            const string OurNamespace = nameof(Microsoft) + "." + nameof(CodeAnalysis) + "." + nameof(ErrorReporting);
+
+            // Be resilient to failing here.  If we can't get a suitable name, just fallback to the standard name we
+            // used to report.
+            try
+            {
+                // walk up the stack looking for the first call from a type that isn't in the ErrorReporting namespace.
+                foreach (var frame in new StackTrace().GetFrames())
+                {
+                    var method = frame.GetMethod();
+                    var methodName = method?.Name;
+                    if (methodName == null)
+                        continue;
+
+                    var declaringTypeName = method?.DeclaringType?.FullName;
+                    if (declaringTypeName == null)
+                        continue;
+
+                    if (declaringTypeName.StartsWith(OurNamespace))
+                        continue;
+
+                    return declaringTypeName + "." + methodName;
+                }
```

Comment:
i've manually stepped through to make sure this works.

Reply:
Wouldn't walking the exception stack trace (`new StackTrace(exception).GetFrames()`) be better, finding a namespace that is a strict subnamespace of `Microsoft.CodeAnalysis`.

Reply:
will do that.

Reply:
> Do we want specific method name, or rather group all issues in the same namespace/type together?

I'd be interested to see how this breaks things down.  I think this approach should give us nice unique issues, without over or underbucketing.  We can always tweak later :)